--- linux-6.13.6-gentoo/drivers/pci/pci-sysfs.c	2025-03-08 08:22:44.686204341 +0100
+++ linux-6.13.7-gentoo/drivers/pci/pci-sysfs.c	2025-03-16 20:57:24.900796726 +0100
@@ -684,6 +684,8 @@
 	struct pci_dev *pdev = to_pci_dev(dev);
 	struct pci_dev *vga_dev = vga_default_device();
 
+	return sysfs_emit(buf, "%u\n", 1);
+
 	if (vga_dev)
 		return sysfs_emit(buf, "%u\n", (pdev == vga_dev));
 
@@ -1592,7 +1594,7 @@
 	struct device *dev = kobj_to_dev(kobj);
 	struct pci_dev *pdev = to_pci_dev(dev);
 
-	if (a == &dev_attr_boot_vga.attr && pci_is_vga(pdev))
+	if (a == &dev_attr_boot_vga.attr && (pci_is_vga(pdev) || (pdev->class >> 8) == PCI_CLASS_DISPLAY_OTHER))
 		return a->mode;
 
 	return 0;